cansat-stm32f446/src/main.rs

```diff
@@ -101,11 +137,28 @@ mod app {
     }
 
     /// Toggles led every second
-    #[task(local = [led])]
+    #[task(priority= 1 , local = [led])]
     fn blink(ctx: blink::Context) {
         let led = ctx.local.led;
         led.toggle();
         defmt::info!("Blink");
         blink::spawn_after(1.secs()).unwrap();
     }
+    #[task(priority=1, local = [delay,bme])]
+    fn bme_measure(ctx: bme_measure::Context) {
+        let bme = ctx.local.bme;
+        let delay = ctx.local.delay;
+        match bme.measure(delay) {
+            Ok(measurements) => {
+                defmt::info!("Relative Humidity = {}%", measurements.humidity);
+                defmt::info!("Temperature = {} deg C", measurements.temperature);
+                defmt::info!("Pressure = {} pascals", measurements.pressure);
+            }
+            Err(_) => {
+                defmt::info!("Could not read bme280 due to error");
+            }
+        }
```

Comment:
We should want to treat errors as exceptional cases, so it's better to just return early and execute the "normal" path of the program. It reduces perceived complexity by minimizing nesting.

Also change defmt::info to defmt::error in the Err case.
```suggestion
        let measurements = match bme.measure(delay) {
            Ok(m) => m,
            Err(e) => {
                defmt::error!("Could not read bme280 measurements: {}", e);
                return;
            }
        };
        defmt::info!("Relative Humidity = {}%", measurements.humidity);
        defmt::info!("Temperature = {} deg C", measurements.temperature);
        defmt::info!("Pressure = {} pascals", measurements.pressure);
```

Normally I'd just use `?` operator to propagate error as much as possible and print the message at the top of the program, but we don't have that much flexibility using rtic without introducing needless complexity:

```rust
// pseudocode

enum TaskError {
    Measure(_)
}

impl defmt::Format for TaskError {
    fn format(&self, fmt: Formatter) {
        match self {
            Error::Measure(e) => write!(fmt, "Could not read bme280 measurements: {}", e)
         }
    }
}

fn try_bme_measure(bme: _, delay: _) -> Result<(), TaskError> {
    let measurements = match bme.measure(delay).map_err(TaskError::Measure)?;
    defmt::info!("Relative Humidity = {}%", measurements.humidity);
    defmt::info!("Temperature = {} deg C", measurements.temperature);
    defmt::info!("Pressure = {} pascals", measurements.pressure);
}

#[task(priority = 1, local = [delay, bme])]
fn bme_measure(ctx: bme_measure::Context) {
    let bme = ctx.local.bme;
    let delay = ctx.local.delay;
    if Err(e) = try_bme_measure(bme, delay) {
        defmt::error!("{}", e);
    }
}
```
Maybe we will do something like this if we decide to move the tasks to the platform-independent `cansat` crate

Reply:
notice, that even though our task could return more errors, we would still have only one `defmt::error!` log to rule them all